test/simple/test-tls-psk-client.js

```diff
@@ -0,0 +1,166 @@
+// Copyright & License details are available under JXCORE_LICENSE file
```

Comment:
Shouldn't there be tests both here and in the server file of trying to use TLS with mixed PSK and non-PSK options just to make sure nothing is stepping on anything else?